test/Table_Tests/src/Common_Table_Operations/Add_Row_Number_Spec.enso

```diff
@@ -84,7 +78,7 @@ add_specs suite_builder setup =
 
             t3 = t.add_row_number order_by=["X"] |> materialize |> _.order_by "row_id"
             Problems.assume_no_problems t3
-            t3.at "Row" . to_vector . should_equal [1, 3, 2, 4, 5]
+            t3.at "Row" . to_vector . should_have_relative_ordering [[1, 3], [2], [4, 5]]
```

Comment:
This relative ordering is according to spec - the elements that are equal on `X` can be in arbitrary order, only elements with differing `X` values need to be monotonic.

We could technically make this more detailed test in in-memory - requiring that the sort is 'stable' - i.e. the order of elements of the same equivalence class is the same as in the input. What do you think @AdRiley ? If you think it's useful I can add it in later PR.